docs/reference/components/sources/exec.cue

```diff
@@ -0,0 +1,227 @@
+package metadata
+
+components: sources: exec: {
+	title: "Exec"
+
+	classes: {
+		commonly_used: false
+		delivery:      "at_least_once"
+		deployment_roles: ["sidecar"]
+		development:   "beta"
+		egress_method: "stream"
+		stateful:      false
+	}
+
+	features: {
+		multiline: enabled: false
+		receive: {
+			from: {
+				service: services.exec
+			}
+
+			tls: enabled: false
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	installation: {
+		platform_name: null
+	}
+
+	configuration: {
+		mode: {
+			description: "The type of exec mechanism."
+			required:    true
+			type: string: {
+				enum: {
+					scheduled: "Scheduled exec mechanism."
+					streaming: "Streaming exec mechanism."
+				}
+				syntax: "literal"
+			}
+		}
+		command: {
+			required:    true
+			description: "The command to be run, plus any arguments required."
+			type: array: {
+				items: type: string: {
+					examples: ["echo", "Hello World!", "ls", "-la"]
```

Comment:
I tried an array but got this error message:-

> components.sources.exec.configuration.command.type.array.items.type.string.examples.0: conflicting values string and ["echo","Hello World!"] (mismatched types string and list):

I tried putting examples in different places but it didn't seem to work. I'll try an experiment with it a bit more but I wanted to pushed the other changes first.

Reply:
Yeah, I think the issue here is that the examples there are examples of the array elements rather than examples of lists of elements. I was able to update the cue config to allow arrays to be specified as examples. I wasn't able to push to this branch though.

Could you apply this diff?

```diff
0cb7faabc - (HEAD -> release, origin/release) Allow specifying examples for config options that take an Array (4 minutes ago) <Jesse Szwedko>
diff --git a/docs/reference.cue b/docs/reference.cue
index d5e2f26b9..5bcaed70a 100644
--- a/docs/reference.cue
+++ b/docs/reference.cue
@@ -383,13 +383,18 @@ _values: {

 #TypeArray: {
        _args: required: bool
+       _type: items: type: string
+
        let Args = _args
+       let Type = _type

        if !Args.required {
                // `default` sets the default value.
                default: [...] | null
        }

+       examples?: [...[...Type.items.type]]
+
        // Set `required` to `true` to force disable defaults. Defaults should
        // be specified on the array level and not the type level.
        items: type: #TypePrimitive & {_args: required: true}
diff --git a/docs/reference/components/sources/exec.cue b/docs/reference/components/sources/exec.cue
index 944d76485..92cce4fa5 100644
--- a/docs/reference/components/sources/exec.cue
+++ b/docs/reference/components/sources/exec.cue
@@ -59,8 +59,8 @@ components: sources: exec: {
                        required:    true
                        description: "The command to be run, plus any arguments required."
                        type: array: {
+                               examples: [["echo", "Hello World!"], ["ls", "-la"]]
                                items: type: string: {
-                                       examples: ["echo", "Hello World!", "ls", "-la"]
                                        syntax: "literal"
                                }
                        }
```

Reply:
Done, thank you for the change.